include/revng/Pipes/Isolate.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/EarlyFunctionAnalysis/ABIDetectionPass.h"
+#include "revng/FunctionIsolation/InvokeIsolatedFunctions.h"
+#include "revng/FunctionIsolation/IsolateFunctions.h"
+#include "revng/Pipes/RevngPipes.h"
+#include "revng/Pipes/RootKind.h"
+
+namespace pipeline {
+
+struct ABIDetectPipe {
+  static constexpr auto Name = "ABIDetect";
+
+  std::vector<ContractGroup> getContract() const { return {}; }
+
+  void registerPasses(llvm::legacy::PassManager &Manager) {
+    Manager.add(new EarlyFunctionAnalysis::ABIDetectionPass());
+  }
+};
+
+struct IsolatePipe {
+  static constexpr auto Name = "Isolate";
+
+  std::vector<ContractGroup> getContract() const {
+    return { ContractGroup(Root,
+                           Exactness::Exact,
+                           0,
+                           Isolated,
+                           0,
+                           InputPreservation::Preserve) };
+  }
+
+  void registerPasses(llvm::legacy::PassManager &Manager) {
+    Manager.add(new IsolateFunctions());
+  }
+};
```

Comment:
i am really considering this, the issue is that it would invert the dependencies and have isolation depend on the pipes library, which is really strange

Reply:
Is it? Isolation depends on LLVM too.

Reply:
yeah, then what would happen is that each library will end up depending on pipes, pretty much

Reply:
I see what you mean, it's not great and it will likely increase compile time. However, adding a new pass it's already quite a pain in the ass. I don't really want to make it worse having to add other files in other directories.
Also, if we want to rename things it sucks.